test/unit/ProductDataTest.php

```diff
@@ -141,6 +142,42 @@ public function testWithers()
         self::assertTrue($product->getCustomAttributes()->equals($expectedCustomAttributes));
     }
 
+    public function testWithMediaGalleryEntrySet()
```

Comment:
Would you mind adding another test case for the deleting images just to prove it still works? @Mostafa1695